// @bit/joshk.react-spinners-css.orbitals/dist/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Orbitals;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _styleModule = _interopRequireDefault(require("./style.module.css"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function Orbitals({
  color,
  className,
  style
}) {
  return /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _classnames.default)(_styleModule.default['lds-orbitals'], className),
    style: _objectSpread({}, style)
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _classnames.default)(_styleModule.default['center']),
    style: {
      background: color
    }
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _classnames.default)(_styleModule.default['inner-spin'])
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _classnames.default)(_styleModule.default['inner-arc'], _styleModule.default['inner-arc_start-a']),
    style: {
      borderColor: color
    }
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _classnames.default)(_styleModule.default['inner-arc'], _styleModule.default['inner-arc_end-a']),
    style: {
      borderColor: color
    }
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _classnames.default)(_styleModule.default['inner-arc'], _styleModule.default['inner-arc_start-b']),
    style: {
      borderColor: color
    }
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _classnames.default)(_styleModule.default['inner-arc'], _styleModule.default['inner-arc_end-b']),
    style: {
      borderColor: color
    }
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _classnames.default)(_styleModule.default['inner-moon-a']),
    style: {
      background: color
    }
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _classnames.default)(_styleModule.default['inner-moon-b']),
    style: {
      background: color
    }
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _classnames.default)(_styleModule.default['outer-spin'])
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _classnames.default)(_styleModule.default['outer-arc'], _styleModule.default['outer-arc_start-a']),
    style: {
      borderColor: color
    }
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _classnames.default)(_styleModule.default['outer-arc'], _styleModule.default['outer-arc_end-a']),
    style: {
      borderColor: color
    }
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _classnames.default)(_styleModule.default['outer-arc'], _styleModule.default['outer-arc_start-b']),
    style: {
      borderColor: color
    }
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _classnames.default)(_styleModule.default['outer-arc'], _styleModule.default['outer-arc_end-b']),
    style: {
      borderColor: color
    }
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _classnames.default)(_styleModule.default['outer-moon-a']),
    style: {
      background: color
    }
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _classnames.default)(_styleModule.default['outer-moon-b']),
    style: {
      background: color
    }
  })));
}

Orbitals.propTypes = {
  /** hex color */
  color: _propTypes.default.string,

  /** class name  */
  className: _propTypes.default.string,

  /** style object */
  style: _propTypes.default.object
};
Orbitals.defaultProps = {
  color: '#7f58af',
  className: '',
  style: {}
};
